### PR TITLE
DRAFT: Replacing server/config.py with commerical server config

### PR DESCRIPTION
**Pull request by corey-nm**
This is just to illustrate differences in config. The full set of changes can be found in #601

### CODE REVIEW

File: src/deepsparse/server/config.py [hunk 1]
```diff
@@ -12,152 +12,73 @@
...
     )
-    integration: str = Field(
-        default="default",
-        description=(
-            "Name of deployment integration that this server will be deployed to "
-            "Currently supported options are None for default inference and "
-            "'sagemaker' for inference deployment with AWS Sagemaker"
-        ),
+
+    endpoint: str = Field(
```
Comment (bfineran) [inline src/deepsparse/server/config.py:46]: should make optional (ie may want to omit in sagemaker mode, or user shouldn't have to worry in general if not absolutely required)

File: src/deepsparse/server/config.py [hunk 2]
```diff
@@ -12,152 +12,73 @@
...
-    workers: str = Field(
-        default=max(1, cpu_architecture().num_available_physical_cores // 2),
-        description=(
-            "The number of maximum workers to use for processing pipeline requests. "
-            "Defaults to the number of physical cores on the device."
-        ),
+
+
+class EndpointConfig(BaseModel):
+    name: str = Field(
```
Comment (bfineran) [inline src/deepsparse/server/config.py:42]: would also prefer to make optional - also what's relationship between name and route

File: src/deepsparse/server/config.py [hunk 3]
```diff
@@ -12,152 +12,73 @@
...
+
+    endpoint: str = Field(
+        description="The path to use for this endpoint. E.g. '/predict'."
     )
 
+    task: str = Field(description="Task this endpoint performers")
+
+    model: str = Field(description="Location of the underlying model to use.")
+
+    batch_size: int = Field(description="The batch size to compile the model for.")
```
Comment (bfineran) [inline src/deepsparse/server/config.py:54]: should default to 1 with dynamic enabled

File: src/deepsparse/server/config.py [hunk 4]
```diff
@@ -12,152 +12,73 @@
...
+
+    accept_multiples_of_batch_size: bool = Field(
```
Comment (bfineran) [inline src/deepsparse/server/config.py:56]: need a default here

File: src/deepsparse/server/config.py [hunk 5]
```diff
@@ -12,152 +12,73 @@
...
+        description="""
```
Comment (bfineran) [inline src/deepsparse/server/config.py:57]: nit: prefer `()` with newline joined strings

File: src/deepsparse/server/config.py [hunk 6]
```diff
@@ -12,152 +12,73 @@
...
+Whether to accept any request with a batch size that is a multiple of `batch_size`.
+
+E.g. if batch_size is 1 and this field is True,
+then the model can accept any batch size.
+    """,
+    )
+
+    bucketing: Optional[Union[ImageSizesConfig, SequenceLengthsConfig]] = Field(
+        default=None,
+        description="What input shapes this model can accept. Must specify at least 1",
```
Comment (bfineran) [inline src/deepsparse/server/config.py:67]: add examples to description

File: src/deepsparse/server/config.py [hunk 7]
```diff
@@ -12,152 +12,73 @@
...
+
+
+class ServerConfig(BaseModel):
+    num_cores: int = Field(
+        description="The number of cores available for model execution. "
+        "Defaults to all available cores.",
+    )
+
+    num_concurrent_batches: int = Field(
+        description="""
```
Comment (bfineran) [inline src/deepsparse/server/config.py:78]: nit: triple quote again

File: src/deepsparse/server/config.py [hunk 8]
```diff
@@ -12,152 +12,73 @@
...
+The number of times to partition the available cores. Each batch will have access to
```
Comment (bfineran) [inline src/deepsparse/server/config.py:79]: let's verify with runtime

File: src/deepsparse/server/config.py [hunk 9]
```diff
@@ -12,152 +12,73 @@
...
+    )
+
+
+class ServerConfig(BaseModel):
+    num_cores: int = Field(
+        description="The number of cores available for model execution. "
+        "Defaults to all available cores.",
+    )
+
+    num_concurrent_batches: int = Field(
```
Comment (bfineran) [inline src/deepsparse/server/config.py:77]: prefer something closer to num_works/num_streams over num_concurrent_batches. additionally we'll need an engine doc on tuning this

**Review by bfineran — CHANGES_REQUESTED** [review]

**Comment by corey-nm** [comment]
Address comments in other PR. going to close this one to clog ppls reviewing less

**Pull request closed without merging**